Core/Core/CourseUpgrade/CourseUpgradeHelper.swift

```diff
@@ -0,0 +1,541 @@
+                        Task { @MainActor in
+                            await self.router.hideUpgradeLoaderView(animated: true)
+                            self.trackUpgradeErrorAction(errorAction: .close, error: error, alertType: self.alertType)
+                            self.hideAlertAction()
+                        }
+                    }
+                )
+            )
+
+            router.presentNativeAlert(
+                title: CoreLocalization.CourseUpgrade.FailureAlert.alertTitle,
+                message: error.localizedDescription,
+                actions: actions
+            )
+        }
+    }
+    
+    private var alertType: UpgradeAlertType {
+        switch upgradeHadler?.state {
+        case .basket:
+            return .basket
+        case .checkout:
+            return .checkout
+        case .payment:
+            return .payment
+        case .verify, .complete:
+            return .execute
+        default:
+            return .unknown
+        }
+    }
+    
+    private func hideAlertAction() {
+        delegate?.hideAlertAction()
+        reset()
+    }
+}
+
+extension CourseUpgradeHelper {
+    public func showLoader(animated: Bool = false, completion: (() -> Void)? = nil) {
+        Task {@MainActor [weak self] in
+            guard let self = self else { return }
+            await self.router.hideUpgradeInfo(animated: false)
+            await self.router.showUpgradeLoaderView(animated: animated)
+            completion?()
+        }
+    }
+    
+    public func removeLoader(
+        success: Bool? = false,
+        shouldRemoveView: Bool? = false,
+        completion: (() -> Void)? = nil
+    ) {
+        self.completion = completion
+        if success == true {
+            helperModel = nil
+        }
+        
+        if shouldRemoveView == true {
+            Task {@MainActor in
+                await router.hideUpgradeLoaderView(animated: true)
+                helperModel = nil
+                
+                if success == true {
+                    trackAndResetOnSuccess()
+                } else {
+                    showError()
+                }
+            }
+        } else if success == false {
+            showError()
+        }
+    }
+}
+
+extension CourseUpgradeHelper {
+    private func showSilentRefreshAlert() {
+        var actions: [UIAlertAction] = []
+
+        actions.append(
+            UIAlertAction(
+                title: CoreLocalization.CourseUpgrade.SuccessAlert.silentAlertRefresh,
+                style: .default
+            ) { [weak self] _ in
+                self?.showDashboardScreen()
+                self?.postSuccessNotification(showLoader: true)
+            }
+        )
+        
+        actions.append(
+            UIAlertAction(
+                title: CoreLocalization.CourseUpgrade.SuccessAlert.silentAlertContinue,
+                style: .default
+            ) { [weak self] _ in
+                self?.reset()
+            }
+        )
+    }
```

Comment:
@saeedbashir It looks like you mistakenly removed the `router.presentNativeAlert(`.

Reply:
oh my bad.